### DISCUSSION TITLE
Add the program to linux package managers?

### DISCUSSION THREAD

**Discussion by joaogelado** [ThioJoe/YT-Spammer-Purge#405] [Ideas]
Can we add the program to package managers like apt, pac-man and others? Would be convenient to the user and updates are managed automatically

**Comment by TechStudent10** [comment]
Is there any way to do that with a Python app. If so, please provide an example.

**Reply by mendelsphotography** [reply]
maybe convert it from phyton to an app?

**Reply by TechStudent10** [reply]
That already happens but it's to a `.exe` file, not an AppImage for example.

**Reply by miquelfire** [reply]
The package just needs to produce the end result of the install(_full).sh file. Some of what the install file does is already done by the package manager (like install Python 3, and other tools that this tool needs). The only thing you'll need is to install the program itself somewhere and allow it to run.

I think the main issue might be where the program will store the config files when installed this way. It seems to want to put the files in the working directory, and not a central location.

**Reply by TechStudent10** [reply]
Like in a dotfile config, so create a folder called `.ytspampurge` or something then place the configs in there.

**Reply by miquelfire** [reply]
The issue is that the program doesn't attempt to make a central location like the user's home folder to store stuff. If install via package a package manager, then the user could end up with 20 .ytspampurge folders because somehow they have 20 folders they start the program in.

**Reply by AirplaneA220** [reply]
he could use a sh script to pull the files from the repository I think, idk if it runs automatically if you do it that way tho

**Reply by joaogelado** [reply]
Glances is a package that i'm pretty sure it is based on python

**Comment by compsup** [comment]
For this to be a possibility the app needs to store config files/temp files in their respective locations. Most likely a dotfile for config, ~/.local/share for data files, etc...

For the technical aspect, what needs to happen is when the program checks for a file (loading, writing, or creating) it needs to check if the system is linux, if it is, create the file in the respective location.

**Reply by TechStudent10** [reply]
In Python, it's easy to find the home directory, so there is no need to check. Plus, storing the configs in a `.youtubespammerpurge/*` on any OS is good by me. However, I think it would be a good idea to ask the user if they want to place their configs in the home directory or in the working directory.

**Reply by joaogelado** [reply]
> In Python, it's easy to find the home directory, so there is no need to check. Plus, storing the configs in a `.youtubespammerpurge/*` on any OS is good by me. However, I think it would be a good idea to ask the user if they want to place their configs in the home directory or in the working directory.

Maybe even system wide, although it would be very inconvenient to change somethings. Maybe an CLI would make the job, but it would take a long time.

**Reply by compsup** [reply]
I would be willing to maintain the package on the AUR, I will post an update when I figured that out. I'm not sure the requirements for package managers so I will respond with valid info in a day or so.

**Reply by TechStudent10** [reply]
> I would be willing to maintain the package on the AUR, I will post an update when I figured that out. I'm not sure the requirements for package managers so I will respond with valid info in a day or so.

What is AUR

**Reply by compsup** [reply]
> > I would be willing to maintain the package on the AUR, I will post an update when I figured that out. I'm not sure the requirements for package managers so I will respond with valid info in a day or so.
>
> What is AUR

https://aur.archlinux.org/
The AUR is the Arch Linux way of packaging non official things, just like how Debian packages non offical things through PPA's

**Reply by TechStudent10** [reply]
Ah okay

**Reply by compsup** [reply]
I have figured out the whole package manager stuff. Currently it works fine, but for "Best practice" the config should be stored preferably in a dotfile on Linux. If no one has any objections I will upload the program to the AUR so all arch users can easily install it from there.

**Reply by TechStudent10** [reply]
> I have figured out the whole package manager stuff. Currently it works fine, but for "Best practice" the config should be stored preferably in a dotfile on Linux. If no one has any objections I will upload the program to the AUR so all arch users can easily install it from there.

pinging @ThioJoe so that he sees this

**Reply by Firecul** [reply]
>  If no one has any objections I will upload the program to the AUR [...]

I'd assume ThioJoe would rather do this himself rather than a 3rd party. 
I do like the idea of it being easy to access but I've never looked into the various ways to see how doable they are in this case (eg Aur, Flatpak and Snap)

**Reply by TechStudent10** [reply]
> I've never looked into the various ways to see how doable they are in this case (eg Aur, Flatpak and Snap)

I saw a guide on how to publish Snap packages for Ubuntu-based distros. I think I saw a section for Python apps.

**Reply by ThioJoe** [reply]
> > If no one has any objections I will upload the program to the AUR [...]
> 
> I'd assume ThioJoe would rather do this himself rather than a 3rd party. I do like the idea of it being easy to access but I've never looked into the various ways to see how doable they are in this case (eg Aur, Flatpak and Snap)

I don't have any plans to do this myself but if someone wants to go make an unofficial one that would just be up to them to maintain i guess

**Reply by sbordeyne** [reply]
For configuration management, I wrote `xdgconfig`, a python library that handles configuration, which is as simple as instanciating a class to have a config object that will read its data from $XDG_CONFIG_HOME on unix system, and from AppData\Roaming on Windows (since it's where the configs should be stored)

I'd advise against using a `.ytspammerpurge` directory in the user's home folder because it increases the amount of clutter in the home directory, is harder to backup (compared to `~/.config` which is a standard directory), it's not configurable, and windows does not hide files and folders with a leading period (which makes the user's directory filled with clutter as soon as you have a couple of apps made for *nix installed)
